docs/user_guide.md

```diff
@@ -0,0 +1,119 @@
+   * - :mod:`moscot.problems.time.LineageProblem`
+     - Estimator for modelling time series single cell data based on :cite:`lange-moslin:23`.
+```
+
+#### Spatial data
+
+```{eval-rst}
+.. list-table::
+   :widths: 15 100
+   :header-rows: 1
+
+   * - Problem
+     - Description
+   * - :mod:`moscot.problems.space.AlignmentProblem`
+     - Class for aligning spatial omics data, based on :cite:`zeira:22`.
+   * - :mod:`moscot.problems.space.MappingProblem`
+     - Class for mapping single cell omics data onto spatial data, based on :cite:`nitzan:19`.
+```
+
+#### Spatiotemporal data
+
+```{eval-rst}
+.. list-table::
+   :widths: 15 100
+   :header-rows: 1
+
+   * - Problem
+     - Description
+   * - :mod:`moscot.problems.spatiotemporal.SpatioTemporalProblem`
+     - Class for analyzing time series spatial single-cell data.
+```
+
+#### Multimodal data
+
+```{eval-rst}
+.. list-table::
+   :widths: 15 100
+   :header-rows: 1
+
+   * - Problem
+     - Description
+   * - :mod:`moscot.problems.cross_modality.TranslationProblem`
+     - Class for integrating single-cell multi-omics data, based on :cite:`demetci-scot:22`.
+```
+
+### Generic problems
+
+```{eval-rst}
+.. list-table::
+   :widths: 15 100
+   :header-rows: 1
+
+   * - Problem
+     - Description
+   * - :mod:`moscot.problems.generic.SinkhornProblem`
+     - Class for solving a :term:`linear problem`.
+   * - :mod:`moscot.problems.generic.GWProblem`
+     - Class for solving a :term:`Gromov-Wasserstein` problem.
+   * - :mod:`moscot.problems.generic.FGWProblem`
+     - Class for solving a :term:`fused Gromov-Wasserstein` problem.
+```
+
+## Scalability
+
+In their original formulation, OT algorithms don't scale to large datasets due to their high computational complexity. Moscot overcomes this limitation by allowing for the use of low-rank solvers. In each `solve` method we have the `rank` parameter, by default $-1$ -- the full rank.
+Whenever possible, it's best to start with the full rank, but when needed, the rank should be set to a positive integer. The higher the rank, the better the full-rank approximation. Hence, one should start with a reasonable high rank, e.g. $5000$. Consecutively decrease the rank if needed due to memory constraints. Note that the scale of $\tau_a$ and $\tau_b$ changes whenever we are in the low-rank setting. While they should be still between $0$ and $1$, empirically they should be set in the range between $0.1$ and $0.5$. See {doc}`/notebooks/examples/solvers/100_linear_problems_basic` and {doc}`/notebooks/examples/solvers/300_quad_problems_basic` on how to use low-rank solutions.
+Another option to use the full rank is to specify the `batch_size` parameter of the `solve` method. It determines the number of rows or columns of the cost matrix to materialize during the {term}`Sinkhorn` iterations. Larger values will require more memory and can be adjusted due to memory constraints as well.
```

Comment:
Can we have this before low-rank? and then say, for linear problems we can do the batch_size, which reduces memory complexity (but slightly increase time complexity).

and then we motivate the low-rank: Whenever time complexity in a linear problem (e.g. TemporalProblem) should be reduced, or memory/time complexity in a quadratic (enumarte / link quadratic problems here) problem should be reduced , we use low-rank OT.